doc/MATH.rst

```diff
@@ -0,0 +1,180 @@
+  by Welch Labs - short and punchy video series
+* `Complex Number Fundamentals <https://www.youtube.com/watch?v=5PcpBw5Hbwo>`__
+  by 3Blue1Brown - a longer lecture-style format
+
+More helpful links:
+
+* `Formula for converting a *cartesian coordinate* to a *polar coordinate* <https://www.engineeringtoolbox.com/converting-cartesian-polar-coordinates-d_1347.html>`__
+
+Core Concept
+''''''''''''
+
+One important fact about complex numbers is that one can think of the same
+complex number in two ways: either as a traditional two-dimensional vector or
+as a length and an angle (measured starting from the positive X-axis). When
+adding two complex numbers, they behave exactly like vectors.
+
+But in contrast to vectors, one can also multiply them together and get a new
+complex number. In this multiplication, their second interpretation becomes
+useful: multiplying two complex numbers means multiplying their lengths and
+adding their angles.
+
+In-Game Examples
+''''''''''''''''
+
+Coordinate System
+^^^^^^^^^^^^^^^^^
+
+Taisei's gameplay takes place entirely on a 2D plane, using an "X" and "Y"
+coordinate system.
+
+.. image:: images/taisei-xy-01.png
+   :width: 300pt
+
+Complex numbers in Taisei represent the "X-axis" (or "real") and "Y-axis" (or
+"imaginary"). This is not exactly a widely-adopted way of thinking about it,
+but as mentioned in *Core Concept*, it allows us to perform operations on the
+coordinate system that wouldn't otherwise be possible using vectors.
+
+Simple Movement
+^^^^^^^^^^^^^^^
+
+Lets take a few concrete examples within Taisei's code. For let's look at a
+piece of movement code for a fairy:
+
+.. code-block:: c
+
+   enemy->move = move_towards(VIEWPORT_W/2.0 + 200.0 * I, 0.035);
+
+The important piece here, for our purposes, is:
+
+.. code-block:: c
+
+   VIEWPORT_W/2.0 + 200.0 * I
+
+What this is doing is specifying a position on the "real" X-axis
+(``VIEWPORT_W/2.0``, which in Taisei means ``480.0 / 2.0`` or `240.0`), and
+then specifying a position on the "imaginary" Y-axis (``200.0 * I``). The ``I``
+here is the same ``i`` described in the videos and the above explanation.
+
+So what we're really looking at here is:
+
+.. code-block:: c
+
+   240.0 (real) + 200.0i (imaginary)
+
+Or "move 240 units on the (real) X-axis, and then 200 units on the (imaginary)
+Y-axis."
+
+This is what's called a **Cartesian Coordinate.** What the function
+``move_towards`` then does is make the enemy sprite/object move towards that
+point on the X/Y axis at a certain rate (defined by ``0.035``).
+
+Simple Danmaku
+^^^^^^^^^^^^^^
+
+Of course, there'd be no point in using complex numbers over a simpler X/Y
+system if it didn't provide significant advantages downstream. So let's look at
+a more complicated danmaku pattern to see why complex numbers are more
+effective for this role.
+
+.. code-block:: c
+
+   cmplx aim = cnormalize(global.plr.pos - enemy->pos);
```

Comment:
Hmm, I would say `cnormalize()`, while very useful, is in itself not a good example of where complex numbers shine, because this operation is exactly analogous to its vector counterpart. The real power of complex numbers is in the multiplication semantics that were mentioned earlier (lengths are multiplied, angles are added) — that's an extremely useful property for rotations. This is acknowledged later in the example (good), so maybe that section is a better place to sing praises to complex numbers.

Overall I think we probably shouldn't try too hard to sell complex numbers as being superior to 2D vectors, just explain how we use them and highlight useful properties as well as similarities to/differences from vectors. I don't feel too strongly about this, though.